src/analyzer.ts

```diff
@@ -48,8 +48,12 @@ export type LazyEdgeMap = Map<string, string[]>;
  */
 export class Analyzer {
   private _context: AnalysisContext;
-  constructor(options: Options) {
-    this._context = new AnalysisContext(options);
+  constructor(options: Options|AnalysisContext) {
+    if (options instanceof AnalysisContext) {
```

Comment:
use `options.constructor.name === 'AnalysisContext'` to work around different instances being passed around terrible npm trees.